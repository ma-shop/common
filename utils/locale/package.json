{
  "name": "@ma-shop/locale",
  "version": "0.0.12",
  "main": "dist/index.js",
  "author": "Tyler Benton",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "lodash": "4.17.15"
  },
  "gitHead": "3e213cdf455e3ac36b88a85e1f7c43d1936fe50d"
}
