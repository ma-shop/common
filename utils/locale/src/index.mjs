// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './locale'
export * from './locales'
