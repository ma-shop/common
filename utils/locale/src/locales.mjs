export const locales = [
  // United States
  {
    locale: 'en-US',
    name: 'United States',
    language: 'en',
    country: 'US',
    alpha2: 'US',
    alpha3: 'USA',
    curreny: 'USD',
    ///# @note https://github.com/moment/moment/issues/3624
    moment: 'en',
    isMarketCountry: true,
  },

  // Taiwan
  {
    locale: 'zh-TW',
    name: '臺灣',
    language: 'zh',
    country: 'TW',
    alpha2: 'TW',
    alpha3: 'TWN',
    curreny: 'TWD',
    moment: 'zh-tw',
    isMarketCountry: true,
  },
  {
    locale: 'zh-Hant-TW',
    name: '臺灣',
    language: 'zh-Hant',
    country: 'TW',
    alpha2: 'TW',
    alpha3: 'TWN',
    curreny: 'TWD',
    moment: null,
    isMarketCountry: false,
  },

  // Canada
  {
    locale: 'en-CA',
    name: 'Canada',
    language: 'en',
    country: 'CA',
    alpha2: 'CA',
    alpha3: 'CAN',
    curreny: 'CAD',
    moment: 'en-ca',
    isMarketCountry: true,
  },
  {
    locale: 'fr-CA',
    language: 'fr',
    name: 'Canada',
    country: 'CA',
    alpha2: 'CA',
    alpha3: 'CAN',
    curreny: 'CAD',
    moment: 'fr-ca',
    isMarketCountry: false,
  },

  // Hong Kong
  {
    locale: 'zh-HK',
    name: '香港',
    language: 'zh',
    country: 'HK',
    alpha2: 'HK',
    alpha3: 'HKG',
    curreny: 'HKD',
    moment: 'zh-hk',
    isMarketCountry: true,
  },
  {
    locale: 'en-HK',
    name: 'Hong Kong - English',
    language: 'en',
    country: 'HK',
    alpha2: 'HK',
    alpha3: 'HKG',
    curreny: 'HKD',
    // There's no en-hk locale in moment. But en-HK formats their dates
    // like the UK.
    moment: 'en-gb',
    isMarketCountry: true,
  },
  {
    locale: 'zh-Hant-HK',
    name: '香港',
    language: 'zh-Hant',
    country: 'HK',
    alpha2: 'HK',
    alpha3: 'HKG',
    curreny: 'HKD',
    // traditional chinese
    moment: 'zh-tw',
    isMarketCountry: false,
  },

  // Singapore
  {
    locale: 'en-SG',
    name: 'Singapore',
    language: 'en',
    country: 'SG',
    alpha2: 'SG',
    alpha3: 'SPG',
    curreny: 'SGD',
    moment: 'en-SG',
    isMarketCountry: true,
  },
  {
    locale: 'zh-SG',
    name: '新加坡',
    language: 'zh',
    country: 'SG',
    alpha2: 'SG',
    alpha3: 'SPG',
    curreny: 'SGD',
    // simplified chinese
    moment: 'zh-cn',
    isMarketCountry: true,
  },
  {
    locale: 'zh-Hans-SG',
    name: '新加坡',
    language: 'zh-Hans',
    country: 'SG',
    alpha2: 'SG',
    alpha3: 'SPG',
    curreny: 'SGD',
    // simplified chinese
    moment: 'zh-cn',
    isMarketCountry: false,
  },

  // Australia
  {
    locale: 'en-AU',
    name: 'Australia',
    language: 'en',
    country: 'AU',
    alpha2: 'AU',
    alpha3: 'AUS',
    curreny: 'AUD',
    moment: 'en-au',
    isMarketCountry: true,
  },

  // United Kingdom
  {
    locale: 'en-GB',
    name: 'United Kingdom',
    language: 'en',
    country: 'GB',
    alpha2: 'GB',
    alpha3: 'GBR',
    curreny: 'GBP',
    moment: 'en-gb',
    isMarketCountry: true,
  },

  // Malaysia
  {
    locale: 'ms-MY',
    name: 'Malaysia',
    language: 'ms',
    country: 'MY',
    alpha2: 'MY',
    alpha3: 'MYS',
    curreny: 'MYR',
    moment: 'ms-my',
    isMarketCountry: true,
  },
  {
    locale: 'en-MY',
    name: 'Malaysia - English',
    language: 'en',
    country: 'MY',
    alpha2: 'MY',
    alpha3: 'MYS',
    curreny: 'MYR',
    moment: 'en',
    isMarketCountry: true,
  },
  {
    locale: 'zh-MY',
    name: '马来西亚',
    language: 'zh',
    country: 'MY',
    alpha2: 'MY',
    alpha3: 'MYS',
    curreny: 'MYR',
    // simplified chinese
    moment: 'zh-cn',
    isMarketCountry: true,
  },
  {
    locale: 'zh-Hans-MY',
    name: '马来西亚',
    language: 'zh-Hans',
    country: 'MY',
    alpha2: 'MY',
    alpha3: 'MYS',
    curreny: 'MYR',
    moment: null,
    isMarketCountry: false,
  },
]
