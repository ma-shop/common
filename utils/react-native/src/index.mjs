// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './get-device-locale'
export * from './interactions'
export * from './is'
export * from './navigation'
export * from './shapes'
