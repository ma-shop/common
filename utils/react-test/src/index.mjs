// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './enzyme'
export * from './react-test'
export * from './render-items'
