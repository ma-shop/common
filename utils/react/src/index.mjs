// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './hoist-static'
export * from './map-props'
export * from './render-if'
export * from './set-ref'
export * from './set-state-async'
export * from './shapes'
export * from './use-previous'
