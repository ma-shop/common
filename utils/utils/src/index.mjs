// @note this is not automated

export * from './delay'
export * from './lodash'
export * from './noop'
export * from './normalize'
export * from './round'
export * from './set-env'
export * from './ui'

// @note this has to be last because jest is weird
export * from './packages'
