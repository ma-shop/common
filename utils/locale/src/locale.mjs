import { find as lodashFind, omit } from 'lodash'

import { locales } from './locales'


const find = (values, _, omittedValues = []) => {
  let obj = _
  // if it's a string assume it's by values
  if (typeof obj === 'string') {
    obj = (item) => Object.values(omit(item, ...omittedValues)).includes(_)
  }
  return lodashFind(values, obj)
}

export class LocaleManager {
  static localeItems = locales

  static locales = LocaleManager.localeItems.map(({ locale }) => locale)

  static get marketLocales () {
    return LocaleManager.localeItems
      .filter(({ isMarketCountry }) => isMarketCountry)
      .map(({ locale }) => locale)
  }

  // stores the current local
  current = 'en-US'

  // expose locales to the class
  locales = LocaleManager.locales

  // expose locales to the class
  marketLocales = LocaleManager.marketLocales

  setLocale (locale) {
    if (!this.locales.includes(locale) && locale !== 'cimode') {
      throw new Error(`Unsupported locale: ${locale}`)
    }

    this.current = locale
  }

  ///# @name parse
  ///# @description parse anything associated with a locale.
  ///# @arg {function, object, array, string} - This accepts the same arguments as the lodash
  ///# [find](https://lodash.com/docs/4.17.15#find) function
  ///# @returns {object} - All the information for that locale
  ///# ```js
  ///# {
  ///#   locale: 'en-US',
  ///#   name: 'United States',
  ///#   language: 'en',
  ///#   country: 'US',
  ///#   alpha2: 'US',
  ///#   alpha3: 'USA',
  ///#   currency: 'USD',
  ///#   moment: 'en',
  ///#   isMarketCountry: true,
  ///# }
  ///# ```
  ///# @example all these examples result in the return object example above
  ///# l.parse('US')
  ///# l.parse({ name: 'United States' })
  ///# l.parse({ currency: 'USD' })
  ///# l.parse([ 'name', 'United States' ])
  ///# // this would return the `en-MY` locale
  ///# l.parse(({ locale }) => locale.includes('MY') && locale.length <= 5)
  parse (obj = this.current, omittedValues) {
    return find(LocaleManager.localeItems, obj, omittedValues)
  }

  /// @name is
  /// @description Takes a locale string and values to see if it's one of the passed countries
  /// @arg {string, array} items - This can be a single value or an array
  /// of values that match the country or locale
  /// @returns {boolean}
  is (...args) {
    // ensures the original array is not modified, and ensures any
    // single values are converted to an array
    const items = [].concat(...args)

    // special cases to check for asian/european countries
    if (items.includes('asia')) {
      items.push('HK', 'SG', 'MY', 'TW')
    } else if (items.includes('europe')) {
      items.push('GB')
    }

    return !!items.find((item) => {
      return Object.values(this.parse(this.current, [ 'moment' ])).includes(item)
    })
  }
}


Object.keys(LocaleManager.localeItems[0]).forEach((key) => {
  Object.defineProperty(LocaleManager.prototype, key, {
    get () {
      // always exclude moment unless the key is moment because it doesn't make any sense
      return this.parse(this.current, key === 'moment' ? [] : [ 'moment' ])[key]
    },
  })
})


export const l = new LocaleManager()
