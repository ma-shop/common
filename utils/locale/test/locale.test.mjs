import { LocaleManager } from '../src/locale'


describe('locale', () => {
  let locale
  beforeEach(() => {
    locale = new LocaleManager()
  })

  test('parse', () => {
    const obj = locale.parse('en-US')

    expect(obj).toMatchSnapshot()
    expect(locale.parse('United States')).toBe(obj)
    expect(locale.parse('US')).toBe(obj)
    expect(locale.parse('USA')).toBe(obj)
    expect(locale.parse('USD')).toBe(obj)
  })

  describe.each(LocaleManager.marketLocales)('%s', (current) => {
    test.each(Object.keys(LocaleManager.localeItems[0]))('%s', (key) => {
      locale.setLocale(current)
      expect(locale[key]).toMatchSnapshot()
    })
  })
})
