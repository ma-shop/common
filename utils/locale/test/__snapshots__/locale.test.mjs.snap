// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locale en-AU alpha2 1`] = `"AU"`;

exports[`locale en-AU alpha3 1`] = `"AUS"`;

exports[`locale en-AU country 1`] = `"AU"`;

exports[`locale en-AU currency 1`] = `"AUD"`;

exports[`locale en-AU isMarketCountry 1`] = `true`;

exports[`locale en-AU language 1`] = `"en"`;

exports[`locale en-AU locale 1`] = `"en-AU"`;

exports[`locale en-AU moment 1`] = `"en-au"`;

exports[`locale en-AU name 1`] = `"Australia"`;

exports[`locale en-CA alpha2 1`] = `"CA"`;

exports[`locale en-CA alpha3 1`] = `"CAN"`;

exports[`locale en-CA country 1`] = `"CA"`;

exports[`locale en-CA currency 1`] = `"CAD"`;

exports[`locale en-CA isMarketCountry 1`] = `true`;

exports[`locale en-CA language 1`] = `"en"`;

exports[`locale en-CA locale 1`] = `"en-CA"`;

exports[`locale en-CA moment 1`] = `"en-ca"`;

exports[`locale en-CA name 1`] = `"Canada"`;

exports[`locale en-GB alpha2 1`] = `"GB"`;

exports[`locale en-GB alpha3 1`] = `"GBR"`;

exports[`locale en-GB country 1`] = `"GB"`;

exports[`locale en-GB currency 1`] = `"GBP"`;

exports[`locale en-GB isMarketCountry 1`] = `true`;

exports[`locale en-GB language 1`] = `"en"`;

exports[`locale en-GB locale 1`] = `"en-GB"`;

exports[`locale en-GB moment 1`] = `"en-gb"`;

exports[`locale en-GB name 1`] = `"United Kingdom"`;

exports[`locale en-HK alpha2 1`] = `"HK"`;

exports[`locale en-HK alpha3 1`] = `"HKG"`;

exports[`locale en-HK country 1`] = `"HK"`;

exports[`locale en-HK currency 1`] = `"HKD"`;

exports[`locale en-HK isMarketCountry 1`] = `true`;

exports[`locale en-HK language 1`] = `"en"`;

exports[`locale en-HK locale 1`] = `"en-HK"`;

exports[`locale en-HK moment 1`] = `"en-gb"`;

exports[`locale en-HK name 1`] = `"Hong Kong - English"`;

exports[`locale en-MY alpha2 1`] = `"MY"`;

exports[`locale en-MY alpha3 1`] = `"MYS"`;

exports[`locale en-MY country 1`] = `"MY"`;

exports[`locale en-MY currency 1`] = `"MYR"`;

exports[`locale en-MY isMarketCountry 1`] = `true`;

exports[`locale en-MY language 1`] = `"en"`;

exports[`locale en-MY locale 1`] = `"en-MY"`;

exports[`locale en-MY moment 1`] = `"en"`;

exports[`locale en-MY name 1`] = `"Malaysia - English"`;

exports[`locale en-SG alpha2 1`] = `"SG"`;

exports[`locale en-SG alpha3 1`] = `"SGP"`;

exports[`locale en-SG country 1`] = `"SG"`;

exports[`locale en-SG currency 1`] = `"SGD"`;

exports[`locale en-SG isMarketCountry 1`] = `true`;

exports[`locale en-SG language 1`] = `"en"`;

exports[`locale en-SG locale 1`] = `"en-SG"`;

exports[`locale en-SG moment 1`] = `"en-SG"`;

exports[`locale en-SG name 1`] = `"Singapore"`;

exports[`locale en-US alpha2 1`] = `"US"`;

exports[`locale en-US alpha3 1`] = `"USA"`;

exports[`locale en-US country 1`] = `"US"`;

exports[`locale en-US currency 1`] = `"USD"`;

exports[`locale en-US isMarketCountry 1`] = `true`;

exports[`locale en-US language 1`] = `"en"`;

exports[`locale en-US locale 1`] = `"en-US"`;

exports[`locale en-US moment 1`] = `"en"`;

exports[`locale en-US name 1`] = `"United States"`;

exports[`locale ms-MY alpha2 1`] = `"MY"`;

exports[`locale ms-MY alpha3 1`] = `"MYS"`;

exports[`locale ms-MY country 1`] = `"MY"`;

exports[`locale ms-MY currency 1`] = `"MYR"`;

exports[`locale ms-MY isMarketCountry 1`] = `true`;

exports[`locale ms-MY language 1`] = `"ms"`;

exports[`locale ms-MY locale 1`] = `"ms-MY"`;

exports[`locale ms-MY moment 1`] = `"ms-my"`;

exports[`locale ms-MY name 1`] = `"Malaysia"`;

exports[`locale parse 1`] = `
Object {
  "alpha2": "US",
  "alpha3": "USA",
  "country": "US",
  "currency": "USD",
  "isMarketCountry": true,
  "language": "en",
  "locale": "en-US",
  "moment": "en",
  "name": "United States",
}
`;

exports[`locale zh-HK alpha2 1`] = `"HK"`;

exports[`locale zh-HK alpha3 1`] = `"HKG"`;

exports[`locale zh-HK country 1`] = `"HK"`;

exports[`locale zh-HK currency 1`] = `"HKD"`;

exports[`locale zh-HK isMarketCountry 1`] = `true`;

exports[`locale zh-HK language 1`] = `"zh"`;

exports[`locale zh-HK locale 1`] = `"zh-HK"`;

exports[`locale zh-HK moment 1`] = `"zh-hk"`;

exports[`locale zh-HK name 1`] = `"香港"`;

exports[`locale zh-MY alpha2 1`] = `"MY"`;

exports[`locale zh-MY alpha3 1`] = `"MYS"`;

exports[`locale zh-MY country 1`] = `"MY"`;

exports[`locale zh-MY currency 1`] = `"MYR"`;

exports[`locale zh-MY isMarketCountry 1`] = `true`;

exports[`locale zh-MY language 1`] = `"zh"`;

exports[`locale zh-MY locale 1`] = `"zh-MY"`;

exports[`locale zh-MY moment 1`] = `"zh-cn"`;

exports[`locale zh-MY name 1`] = `"马来西亚"`;

exports[`locale zh-SG alpha2 1`] = `"SG"`;

exports[`locale zh-SG alpha3 1`] = `"SGP"`;

exports[`locale zh-SG country 1`] = `"SG"`;

exports[`locale zh-SG currency 1`] = `"SGD"`;

exports[`locale zh-SG isMarketCountry 1`] = `true`;

exports[`locale zh-SG language 1`] = `"zh"`;

exports[`locale zh-SG locale 1`] = `"zh-SG"`;

exports[`locale zh-SG moment 1`] = `"zh-cn"`;

exports[`locale zh-SG name 1`] = `"新加坡"`;

exports[`locale zh-TW alpha2 1`] = `"TW"`;

exports[`locale zh-TW alpha3 1`] = `"TWN"`;

exports[`locale zh-TW country 1`] = `"TW"`;

exports[`locale zh-TW currency 1`] = `"TWD"`;

exports[`locale zh-TW isMarketCountry 1`] = `true`;

exports[`locale zh-TW language 1`] = `"zh"`;

exports[`locale zh-TW locale 1`] = `"zh-TW"`;

exports[`locale zh-TW moment 1`] = `"zh-tw"`;

exports[`locale zh-TW name 1`] = `"臺灣"`;
